ui/apps/dashboard/src/components/Insights/QueryHelperPanel/StoredQueriesContext.tsx

```diff
@@ -101,6 +103,22 @@ export function StoredQueriesProvider({ children, tabManagerActions }: StoredQue
     [beDeleteQuery, refetchSavedQueries, tabManagerActions]
   );
 
+  const shareQuery = useCallback(
+    async (queryId: string) => {
+      const result = await beShareQuery({ id: queryId });
+      if (result.ok) {
+        // TODO: This often leads to double-fetching, but it's currently needed because the "InsightsQueryStatement"
+        // __typename does not exist and does not auto-refetch if the list was previously empty. We need to make sure
+        // that we have a consistent type name to match on regardless of existing saved queries.
```

Comment:
can you elaborate on this? is it because we don't have `__typename` yet before hitting the backend once?